eng/CodeAnalysis.src.globalconfig

```diff
@@ -467,6 +467,7 @@ dotnet_diagnostic.CA1858.severity = warning
 
 # CA1859: Use concrete types when possible for improved performance
 dotnet_diagnostic.CA1859.severity = warning
+dotnet_diagnostic.CA1859.api_surface = private, internal
```

Comment:
@buyaa-n, when I built locally, this didn't raise any additional diagnostics, but from what I can tell we are using a version that has the latest changes.  Are we sure it's working as expected?  It's of course possible there's nothing else, it's just surprising.

Reply:
Unless I'm misunderstanding what this is supposed to do or how to enable it, it doesn't appear to be working correctly.  I tried adding:
```C#
private IList<int> _list;
```
to a class and:
```C#
_list = new List<int>();
_list.Add(42);
```
to that class' ctor.  It raised a diagnostic.  Then I changed `private` to `internal` and built again... no diagnostic.

Reply:
@geeknoid, should this be working, or do I need to do something differently?

Reply:
it should be `dotnet_code_quality.CA1859.api_surface = private, internal`

Reply:
> it should be dotnet_code_quality

Ah, code_quality instead of diagnostic. Ok.  It's strange that a different prefix is used for controlling its severity than from controlling its other settings.